sdk/keyvault/keyvault-secrets/src/identifier.ts

```diff
@@ -0,0 +1,71 @@
+import { parseKeyvaultIdentifier } from "./core/utils";
+
+/**
+ * Represents a KeyVault identifier and its parsed contents.
+ */
+export interface ParsedKeyVaultIdentifier {
+  /**
+   * The type of resource under KeyVault that this identifier is referring to.
+   */
+  collection: "keys" | "secrets" | "certificates";
+
+  /**
+   * The originally received identifier.
+   */
+  id: string;
+
+  /**
+   * The KeyVault Key unique identifier (an URl).
+   */
+  vaultUrl: string;
+
+  /**
+   * The version of KeyVault Key. May be undefined.
+   */
+  version?: string;
+
+  /**
+   * The name of the KeyVault Key.
+   */
+  name: string;
+}
+
+/**
+ * Parser of the KeyVaultIdentifier for the KeyVault Keys Client.
+ */
+export class KeyVaultIdentifier implements ParsedKeyVaultIdentifier {
+  /**
+   * The type of resource under KeyVault that this identifier is referring to.
+   */
+  collection: "keys" | "secrets" | "certificates";
+
+  /**
+   * The originally received identifier.
+   */
+  public id: string;
+
+  /**
+   * The KeyVault Key unique identifier (an URl).
+   */
+  public vaultUrl: string;
+
+  /**
+   * The version of KeyVault Key. May be undefined.
+   */
+  public version?: string;
+
+  /**
+   * The name of the KeyVault Key.
+   */
+  public name: string;
+
+  constructor(url: string) {
+    const coreParsedIdentifier = parseKeyvaultIdentifier("secrets", url);
+
+    this.collection = "keys";
```

Comment:
Another hardcoded instance of "keys" for secrets. Also, we should allow that this could be a deleted cert, key, or secret, which would be collection "deleted<whatever>".

Reply:
...that is to say, we should parse out the collection and, perhaps, assert it.

Reply:
This was mostly due to copy and paste, but you've pointed out to the deleted collection, which is something I wasn't aware of. Thank you!

Reply:
I'll add tests for this too.

Reply:
Are you sure that the deleted collection is a thing? I logged each use case of our current tests using the generated identifier parser and none of them passed in a certificate located in the deleted collection.